gax-java/gax/src/test/java/com/google/api/gax/util/TimeConversionUtilsTest.java

```diff
@@ -69,4 +69,44 @@ void testToThreetenTimeInstant_validInput_succeeds() {
         jtInstant.toEpochMilli(), TimeConversionUtils.toThreetenInstant(jtInstant).toEpochMilli());
     assertNull(TimeConversionUtils.toThreetenInstant(null));
   }
+
+  @Test
+  void testToThreeteenInstant_bigInput_doesNotOverflow() {
+    // defaults to MAX_SECONDS plus the max value of long for the nanos part
```

Comment:
I'm not too familiar with this. What do you mean by `plus the max value of long for nanos`?

I see that the MAX is `1000000000-12-31T23:59:59.999999999Z`. Does this mean that this is essentially where jtInstant.getNano() and jtInstant.getEpochSecond() are both Interger.MAX_VALUE?

Reply:
Instants and Durations have two components: the seconds part and the nanos part, both being `long` for `Duration`, and `long` for seconds + `int` for nanos [in Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html).

> Does this mean that this is essentially where jtInstant.getNano() and jtInstant.getEpochSecond() are both Interger.MAX_VALUE?

Yes, but with `Long.MAX_VALUE` for the seconds part instead.

Reply:
Ok gotcha. I couldn't make sense of where the overflow was originally coming from since MAX_VALUE itself shouldn't trigger an overflow. If I'm understanding this correctly, it was triggered from the old way of how we were converting.

The `toNanos()` method was limited to Long.MAX_VALUE which has a limit of 292 years. Since Instant.MAX_VALUE has Long.MAX_VALUE for both seconds and nanos, it wouldn't be able to fit into `toNanos()` into a Long.MAX_VALUE as adding all the seconds + nanos would easily overfill.

Reply:
> If I'm understanding this correctly, it was triggered from the old way of how we were converting.

Exactly.

> it wouldn't be able to fit into toNanos() into a Long.MAX_VALUE as adding all the seconds + nanos would easily overfill

Yes. I don't think we will ever deal with durations instants that last longer than a few hours but it's good to support the theoretical limits of these classes, even because a user may just try a max value and that may break the library.